File: snippets/run_actor.py
"""Run the Hotels.com Reviews Scraper on Apify and print the results.

Install:  pip install apify-client
Docs:     https://apify.com/factden/hotels-com-reviews-scraper
"""

from apify_client import ApifyClient

# Get your token from https://console.apify.com/settings/integrations
client = ApifyClient("<YOUR_APIFY_TOKEN>")

run_input = {
    "hotelUrls": [
        "https://www.hotels.com/ho119566/",
    ],
    "maxReviewsPerHotel": 100,
    "sortBy": "newest",
}

# Start the actor and wait for it to finish
run = client.actor("factden/hotels-com-reviews-scraper").call(run_input=run_input)

# Iterate the resulting dataset (one row per review)
for row in client.dataset(run["defaultDatasetId"]).iterate_items():
    print(f"{row.get('overallRating')}/10  {row.get('hotelName')}")
    if row.get("ownerResponse"):
        print("   ↳ hotel responded")

File: snippets/run_actor.js
// Run the Hotels.com Reviews Scraper on Apify and print the results.
//
// Install:  npm install apify-client
// Docs:     https://apify.com/factden/hotels-com-reviews-scraper

import { ApifyClient } from 'apify-client';

// Get your token from https://console.apify.com/settings/integrations
const client = new ApifyClient({ token: '<YOUR_APIFY_TOKEN>' });

const input = {
    hotelUrls: [
        'https://www.hotels.com/ho119566/',
    ],
    maxReviewsPerHotel: 100,
    sortBy: 'newest',
};

// Start the actor and wait for it to finish
const run = await client.actor('factden/hotels-com-reviews-scraper').call(input);

// Fetch the resulting dataset (one row per review)
const { items } = await client.dataset(run.defaultDatasetId).listItems();
for (const row of items) {
    console.log(`${row.overallRating}/10  ${row.hotelName}`);
    if (row.ownerResponse) {
        console.log('   ↳ hotel responded');
    }
}
